### ISSUE TITLE
Allow for movement on the Z-axis

### ISSUE THREAD

**Issue by AbdelHajou** [han-yaeger/yaeger#178]
It should be easier to change the direction of a sprite. Currently you need to supply character sprites for every direction (walking left, walking right etc.). It would be easier to be able to use a single sprite and flip it according to the direction the character is moving.

Another enhancement that has to do with movement on the Z-axis is changing the depth of a sprite. Sprites are currently drawn in the order that they were added to the scene, so sometimes a sprites is unintentionally drawn on top of another sprite (e.g. a player being drawn behind an obstacle).

**Comment by SimonVreman** [comment]
I came across this when discovering #190 and needed a solution so I made an interface that utilizes `@OnActivation` to align entities on the z-axis. My solution is as below.

```java
public interface AlignZAxis {

    @OnActivation
    default void alignZAxis() {
        if (this.getNode().isPresent()) {
            this.getNode().get().setViewOrder(getZIndex());
        }
    }

    double getZIndex();

    Optional<? extends Node> getNode();
}
```

This could be expanded on, but it works as it is. Note that a higher ZIndex value means a lower priority, so the lowest ZIndex is drawn on top. A bit weird as it's the inverse of CSS ZIndex. More in depth documentation can be [found in the JavaFX docs](https://openjfx.io/javadoc/16/javafx.graphics/javafx/scene/Node.html#setViewOrder(double)).

Something similar can be done for image flipping, based on something like this.

```java
public interface Flippable {

    @OnActivation
    default void flipImage() {
        if (this.getNode().isPresent()) {
            var node = this.getNode().get();
            var flip = node.getScaleX() * -1;
            node.setScaleX(flip);
        }
    }

    Optional<? extends Node> getNode();
}
```

This flips the image when activated and every time `flipImage()` is called.

**Comment by meronbrouwer** [comment]
Good work again @SimonVreman I will implement the viewOrder solution as a fix for bug #190. Entities from a TileMap will default to a high number and the user will be able to set it to a desired value.

Regarding your second remark (about the flipImage()), I'm not yet sure if we should expose this through the API. There a more translations possible for nodes, but exposing too many of those will make the API too large. Students will only have a few weeks to get familiar with the engine and implement a game. A larger API might limit their focus on actually creating a game.

**Comment by SimonVreman** [comment]
Yes that is true, including flipped versions in the sheet resource as in the tutorial with Hanny works just fine as well and doesn't need to complicate the engine further.
